### PR TITLE
[Inductor] Replace functools.reduce union calls with set unions

### PR DESCRIPTION
**Pull request by mlazos**
This improves compilation speed another 30%

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

### CODE REVIEW

File: torch/_inductor/scheduler.py [hunk 1]
```diff
@@ -534,7 +533,7 @@ def get_first_name(self) -> str:
 
     @cache_on_self
     def get_names(self) -> Set[str]:
-        return functools.reduce(set.union, [x.get_names() for x in self.snodes])
+        return set.union(*[x.get_names() for x in self.snodes])
```
Comment (lezcano) [inline torch/_inductor/scheduler.py:536]: Even better
```suggestion
        return set.union(x.get_names() for x in self.snodes)
```
same in the other two instances below.

File: torch/_inductor/scheduler.py [hunk 2]
```diff
@@ -1155,8 +1152,10 @@ def can_fuse(self, node1: BaseSchedulerNode, node2: BaseSchedulerNode):
             and not node2.is_template()
         ):
             return False
+
         if node2.get_names() & node1.recursive_predecessors:
             return False  # node2 must go before node1
+
```
Comment (lezcano) [inline torch/_inductor/scheduler.py:1158]: Two stray new lines?

**Review by lezcano — APPROVED** [review]
Cool! Just a couple comments.

File: torch/_inductor/scheduler.py (hunk 1 above)
Comment (lezcano) [reply torch/_inductor/scheduler.py:536]: ok, so `set.union` doesn't accept generators. You could still do `*(x.get...)`, but yeah.

File: torch/_inductor/scheduler.py (hunk 2 above)
Comment (mlazos) [reply torch/_inductor/scheduler.py:1158]: yeah I thought these helped with readability

**Pull request closed without merging**